src/app/src/main/kotlin/org/imperial/mrc/hint/security/SecurityConfig.kt

```diff
@@ -38,14 +39,31 @@ class Pac4jConfig {
 @Component
 class Session(private val webContext: WebContext, private val pac4jConfig: Config) {
 
-    fun getId(): String = pac4jConfig.sessionStore.getOrCreateSessionId(webContext)
+    companion object {
+        private const val SNAPSHOT_ID = "snapshot_id"
+    }
 
     fun getUserProfile(): CommonProfile {
         val manager = ProfileManager<CommonProfile>(webContext)
         val profiles = manager.getAll(true)
         return profiles.singleOrNull() ?: CommonProfile().apply {
             id = GUEST_USER
         }
+    }
+
+    fun getSnapshotId() :String {
```

Comment:
So I *think* this is all implicitly under test because otherwise the integration tests that upload files that depend on the shape file wouldn't work (e.g. if it was generating a new shapshot id every time rather than correctly reading it from the session). Could add some unit tests for these methods but would be a faff to mock the session store, so I'm happy with implicit tests if you are!

Reply:
(Go ahead and merge if so)

Reply:
Yeah I noticed the same thing and came to the same conclusio!